src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -286,30 +290,7 @@ private void validateArrayValue(
       ++depth;
       validateDocDepth(limits, depth);
 
-      if (arrayValue.size() > limits.maxArrayLength()) {
```

Comment:
Moved out of full-doc validation, as it only should apply to indexable array fields.